backend/src/twitter-feed/twitter-feed.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '../config/config.service';
+import { FeedService } from '../feeds/feed.service';
+const config = new ConfigService();
+const Twitter = require('twitter')
+
+const twitter = new Twitter({
+    consumer_key: config.twitterConsumerKey,
+    consumer_secret: config.twitterConsumerSecret,
+    access_token_key: config.twitterTokenKey,
+    access_token_secret: config.twitterTokenSecret,
+})
+
+@Injectable()
+export class TwitterFeedService {
+
+    constructor(
+        private readonly feedService: FeedService,
+        private readonly logger: Logger) {
+        this.logger = new Logger('TwitterFeedService');
+    }
+
+    onModuleInit() {
+        this.fetchTwitterFeeds();
+    }
+
+    async fetchTwitterFeeds() {
```

Comment:
the convention is that when you `fetch` something, you expect it to have a promise back to you.
For example:

`window.fetch("https://vantaa.dev.fi/mobile-newstand/api/feed/fb")` It is more natural to expect that this return a Promise and have all of the consumption and error handling dictated by you.

So I suggest you rename this function to `fetchAndSaveToDbTwitterFeed` or just inline this in your main function.

Reply:
On second note, this seems like a main function for this service.
Then maybe a more fitting naming.

Clean API prescribes that you can know what happens by just looking at the function name